velox/functions/prestosql/aggregates/ChecksumAggregate.cpp

```diff
@@ -0,0 +1,225 @@
+    vector->resize(numGroups);
+    uint64_t* rawNulls = getRawNulls(vector);
+
+    for (int32_t i = 0; i < numGroups; ++i) {
+      char* group = groups[i];
+      if (isNull(group)) {
+        vector->setNull(i, true);
+      } else {
+        clearNull(rawNulls, i);
+        vector->set(i, StringView(value<char>(group), sizeof(int64_t)));
+      }
+    }
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    const auto& arg = args[0];
+    auto hasher = getPrestoHasher(arg->type());
+    BufferPtr hashes =
+        AlignedBuffer::allocate<int64_t>(rows.end(), arg->pool());
+    hasher->hash(arg, rows, hashes);
+    auto rawHashes = hashes->as<int64_t>();
+
+    rows.template applyToSelected([&](vector_size_t row) {
+      if (arg->isNullAt(row)) {
+        computeHashForNull(groups[row]);
+      } else {
+        computeHash(groups[row], rawHashes, row);
+      }
+    });
+  }
+
+  void addIntermediateResults(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    VELOX_CHECK_EQ(args[0]->encoding(), VectorEncoding::Simple::FLAT);
+
+    auto vector = args[0]->asUnchecked<FlatVector<StringView>>();
+    auto rawValues = vector->rawValues();
+
+    rows.applyToSelected([&](vector_size_t i) {
+      *value<int64_t>(groups[i]) = *(getHashValue(rawValues[i]));
+    });
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    const auto& arg = args[0];
+    auto hasher = getPrestoHasher(arg->type());
+    BufferPtr hashes =
+        AlignedBuffer::allocate<int64_t>(rows.end(), arg->pool());
+    hasher->hash(arg, rows, hashes);
+    auto rawHashes = hashes->as<int64_t>();
+
+    rows.template applyToSelected([&](vector_size_t row) {
+      if (arg->isNullAt(row)) {
+        computeHashForNull(group);
+      } else {
+        computeHash(group, rawHashes, row);
+      }
+    });
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /**/) override {
+    VELOX_CHECK_EQ(args[0]->encoding(), VectorEncoding::Simple::FLAT);
+
+    auto vector = args[0]->asUnchecked<FlatVector<StringView>>();
+    auto rawValues = vector->rawValues();
+    int64_t result = 0;
+    if (vector->mayHaveNulls()) {
+      rows.applyToSelected([&](vector_size_t i) {
+        if (!vector->isNullAt(i)) {
+          result += *(getHashValue(rawValues[i]));
+        }
+      });
+    } else {
+      rows.applyToSelected(
+          [&](vector_size_t i) { result += *(getHashValue(rawValues[i])); });
+    }
+
+    *value<int64_t>(group) = result;
+  }
+
+ private:
+  FOLLY_ALWAYS_INLINE void
+  computeHash(char* group, const int64_t* hashes, const vector_size_t& row) {
```

Comment:
Looks like this function could take a group and a single hash: `(char* group, int64_t hash)`